bug(admin): changelist crashes with 500 when listDisplay includes a ForeignKey to a model that has its own unloaded ForeignKey fields

## Description

The admin changelist view (`/admin/<model>/`) crashes with a `500 Internal Server Error` when a model's `listDisplay` includes a `ForeignKey` field, and the related model itself has `ForeignKey` fields of its own (i.e. nested FK relations).

## Steps to Reproduce

1. Define a model with a `ForeignKey` to another model that also has `ForeignKey` fields:

```typescript
// Unit has its own FK fields (e.g. owner, parent)
class Unit extends Model {
  id = new AutoField({ primaryKey: true });
  name = new CharField({ maxLength: 100 });
  owner = new ForeignKey<User>("User", { onDelete: OnDelete.SET_NULL, blank: true });
}

class Member extends Model {
  id = new AutoField({ primaryKey: true });
  email = new CharField({ maxLength: 100 });
  unit = new ForeignKey<Unit>("Unit", { onDelete: OnDelete.CASCADE, relatedName: "members" });
}
```

2. Register `Member` in the admin with `unit` in `listDisplay`:

```typescript
class MemberAdmin extends ModelAdmin {
  override listDisplay = ["id", "email", "unit"];
}
```

3. Navigate to `/admin/member/` in the browser.

## Expected Behavior

The changelist renders correctly, showing the related `Unit` as a readable value (e.g. its string representation or primary key).

## Actual Behavior

```
[ERROR] GET /admin/member/: Error: Related object 'owner' not fetched. Call .fetch() first or use selectRelated().
    at ForeignKey.get (https://jsr.io/@alexi/db/0.41.0/fields/relations.ts:229:13)
    at Proxy.toObject (https://jsr.io/@alexi/db/0.41.0/models/model.ts:797:26)
    at Proxy.toJSON (https://jsr.io/@alexi/db/0.41.0/models/model.ts:818:17)
    at JSON.stringify (<anonymous>)
    at formatValue (https://jsr.io/@alexi/admin/0.41.0/views/changelist_views.ts:56:46)
    at https://jsr.io/@alexi/admin/0.41.0/views/changelist_views.ts:615:23
← 500 Internal Server Error
```

## Root Cause

The bug is a chain of three issues in `src/admin/views/changelist_views.ts`:

### Step 1 — `selectRelated` loads top-level FKs on the queried model

```typescript
// changelist_views.ts ~line 457
const fkFieldNames = fields
  .filter((f) => f.type === "ForeignKey" || f.type === "OneToOneField")
  .map((f) => f.name);
const qs = fkFieldNames.length > 0
  ? baseQs.selectRelated(...fkFieldNames)
  : baseQs;
```

This loads `member.unit` (a `Unit` instance) — correct so far.

### Step 2 — `safeGetFieldValue` returns the live model instance

```typescript
// changelist_views.ts ~line 564
allData = pageRows.map((row) => {
  const obj: Record<string, unknown> = {};
  for (const f of fields) {
    obj[f.name] = safeGetFieldValue(r[f.name]);
  }
  return obj;
});

function safeGetFieldValue(v: unknown): unknown {
  // ...
  if (typeof obj["isLoaded"] === "function") {
    const fk = obj as { isLoaded(): boolean; get(): unknown; id: unknown };
    return fk.isLoaded() ? fk.get() : fk.id;  // returns the full Unit instance!
  }
  // ...
}
```

When `unit.isLoaded()` is `true`, `safeGetFieldValue` returns `fk.get()` — the **live `Unit` model instance** — and stores it in `allData["unit"]`.

### Step 3 — `formatValue` calls `JSON.stringify` on the live model instance

```typescript
// changelist_views.ts line 56
function formatValue(value: unknown): string {
  // ...
  if (typeof value === "object") return JSON.stringify(value);  // ← CRASH
  // ...
}
```

`JSON.stringify` internally traverses the `Unit` instance's fields (via `toJSON()`), which encounters `unit.owner` — a `ForeignKey<User>` that was **never loaded** (only the top-level `unit` FK on `Member` was `selectRelated`'d). Calling `.get()` on an unloaded FK throws `Error: Related object 'owner' not fetched.`

## Fix

`safeGetFieldValue` should **not** return a live model instance. When a FK is loaded, it should be reduced to a display-friendly primitive before being placed in `allData`. The two reasonable options are:

**Option A** — Return the FK's primary key (safest, always works):
```typescript
return fk.isLoaded() ? (fk.get() as Model).pk ?? fk.id : fk.id;
```

**Option B** — Return the FK's string representation (more user-friendly):
```typescript
return fk.isLoaded() ? String(fk.get()) : fk.id;
```

**Option C** — Recursively serialize the loaded instance to a plain object with `safeGetFieldValue` applied to each of its own fields (most complete, but complex).

The simplest correct fix is **Option A or B**: reduce the FK value to a primitive in `safeGetFieldValue` so that `formatValue` never receives a live model instance.

Additionally, `formatValue` itself could be made more defensive:

```typescript
function formatValue(value: unknown): string {
  if (value === null || value === undefined) return "-";
  if (typeof value === "boolean") return value ? "Yes" : "No";
  if (value instanceof Date) return value.toLocaleString();
  if (typeof value === "object") {
    // Guard: never call JSON.stringify on a live Model instance
    if (typeof (value as Record<string, unknown>)["toJSON"] === "function") {
      return String((value as Record<string, unknown>)["pk"] ?? value);
    }
    return JSON.stringify(value);
  }
  return String(value);
}
```

## Affected Files

- `src/admin/views/changelist_views.ts` — `safeGetFieldValue` (line ~71) and `formatValue` (line ~52)

## Environment

- Alexi version: 0.41.0
- Reproduced in: `@alexi/admin` changelist view with any model that has a `ForeignKey` in `listDisplay` where the related model itself contains `ForeignKey` fields

--- a/src/admin/views/changelist_views.ts
+++ b/src/admin/views/changelist_views.ts
@@ -61,8 +61,10 @@ function formatValue(value: unknown): string {
  * Safely extract a display value from a model field instance.
  *
  * - For regular fields: calls `.get()` to retrieve the value.
- * - For ForeignKey fields that are loaded: calls `.get()` on the related
- *   instance (which itself has field accessors).
+ * - For ForeignKey fields that are loaded: returns the related object's
+ *   primary key (a primitive) rather than the live model instance, to avoid
+ *   cascading serialization errors when the related model itself has unloaded
+ *   ForeignKey fields.
  * - For ForeignKey fields that are NOT loaded (e.g. not in listDisplay /
  *   selectRelated was not called for this field): falls back to the raw FK ID
  *   via `.id` instead of throwing.
@@ -78,7 +80,15 @@ function safeGetFieldValue(v: unknown): unknown {
       get(): unknown;
       id: unknown;
     };
-    return fk.isLoaded() ? fk.get() : fk.id;
+    if (!fk.isLoaded()) return fk.id;
+    // Return the related object's primary key (a primitive) rather than the
+    // live model instance. Passing the full instance to formatValue would cause
+    // JSON.stringify to traverse its own FK fields, which may not be loaded,
+    // resulting in a "Related object not fetched" error.
+    const related = fk.get() as Record<string, unknown> | null;
+    if (related === null || related === undefined) return fk.id;
+    if (typeof related["pk"] !== "undefined") return related["pk"];
+    return fk.id;
   }
   // Regular field: call .get()
   if (typeof obj["get"] === "function") {

